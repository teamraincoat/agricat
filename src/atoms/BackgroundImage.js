import React from 'react';
import { StyleSheet, ImageBackground } from 'react-native';
import { wp, hp } from '../styles/metrics';

export default function BackgroundImage({
  src,
  backgroundStyle,
  imageStyle,
  children,
}) {
  return (
    <ImageBackground
      style={[styles.imgBackground, backgroundStyle]}
      resizeMode="cover"
      imageStyle={imageStyle}
      source={src}>
      {children}
    </ImageBackground>
  );
}

const styles = StyleSheet.create({
  imgBackground: {
    width: wp(100),
    height: hp(100),
  },
});
