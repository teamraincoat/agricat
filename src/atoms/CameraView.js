/* eslint-disable global-require */
import React from 'react';
import {
  View,
  StyleSheet,
  Platform,
  Alert,
} from 'react-native';
import { CameraScreen, CameraType } from 'react-native-camera-kit';
import ReactNativeBlobUtil from 'react-native-blob-util';
import { hp, wp } from '../styles/metrics';

const CameraView = ({ setIsCameraVisible, selectedFiles, setSelectedFiles }) => {
  const onBottomButtonPressed = async (event) => {
    const captureImage = event.captureImages[0]?.uri;
    if (event?.type === 'left') {
      setIsCameraVisible(false);
      return;
    }
    handleCaptureImage(captureImage);
    setIsCameraVisible(false);
  };

  const handleCaptureImage = async (captureImage) => {
    const selectedImage = [];
    try {
      const imageData = await ReactNativeBlobUtil.fs.stat(captureImage);
      const filePrefix = 'file://';
      const re = /(?:\.([^.]+))?$/;
      const ext = re.exec(imageData)[1];
      const imagePath = filePrefix.concat(imageData.path);
      console.log('imagePath--->>', imagePath);
      console.log('imageData>>', imageData);
      const realURI = Platform.select({
        android: imagePath,
        // ios: decodeURI(res.uri),
      });
      const base64Image = await ReactNativeBlobUtil.fs.readFile(
        realURI,
        'base64',
      );
      selectedImage.push({
        name: imageData.filename,
        uri: base64Image,
        size: imageData.size,
        type: ext,
      });
      setSelectedFiles(selectedFiles.concat(selectedImage));
    } catch {
      Alert.alert('Error leyendo la imagen.');
    }
  };

  return (
    <View style={localStyles.container}>
      <CameraScreen
        actions={{ rightButtonText: 'Done', leftButtonText: 'Cancelar' }}
        onBottomButtonPressed={onBottomButtonPressed}
        cameraType={CameraType.Back}
        focusMode="on"
        zoomMode="off"
        captureButtonImage={require('../assets/capture.png')}
        // captureButtonImageStyle={{ width: 50, height: 50 }}
        hideControls={false}
        showCapturedImageCount={false}
        saveToCameraRoll={false}
      />
    </View>
  );
};

const localStyles = StyleSheet.create({
  container: {
    height: hp(100),
    width: wp(100),
  },
});

export default CameraView;
