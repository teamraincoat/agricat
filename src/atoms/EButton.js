import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';

const EButton = (props) => {
  return (
    <TouchableOpacity style={styles.button} onPress={props.onClick}>
      <Text style={styles.text}>{props.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: '#009999',
    color: '#ffffff',
    padding: 10,
    marginTop: 16,
  },
  text: {
    color: 'white',
    fontWeight: 'bold',
  },
});

export default EButton;
