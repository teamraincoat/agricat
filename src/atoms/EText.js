import React from 'react';
import {Text, StyleSheet} from 'react-native';

const EText = props => {
  return <Text style={[localStyles.text, props.style]} {...props}>{props.children}</Text>;
};

const localStyles = StyleSheet.create({
  text: {
    // color: '#111825',
    // fontSize: 18,
    // textAlign: 'center',
  },
});

export default EText;
