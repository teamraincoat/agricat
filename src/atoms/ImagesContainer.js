import React, { useCallback } from 'react';
import {
  View,
  TouchableOpacity,
  PixelRatio,
  StyleSheet,
  Image,
} from 'react-native';
import DeleteIcon from '../assets/icons/DeleteIcon';

const ImagesContainer = (props) => {
  const { selectedFileImages, setSelectedImages } = props;

  const removeImage = useCallback(
    (imageName) => {
      const newImages = selectedFileImages.filter(
        (item) => item.name !== imageName,
      );
      setSelectedImages(newImages);
    },
    [selectedFileImages],
  );
  return (
    <View style={localStyles.imageContainer}>
      <View style={localStyles.imageInnerContainer}>
      <Image
        source={{ uri: `data:image/png;base64,${selectedFileImages[0].uri}` }}
        style={localStyles.imageStyle}
      />
      <TouchableOpacity
        style={localStyles.deleteIcon}
        onPress={() => removeImage(selectedFileImages[0].name)}>
        <DeleteIcon />
      </TouchableOpacity>
    </View>
    </View>
  );
};

const localStyles = StyleSheet.create({
  imageContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'flex-start',
    marginVertical: PixelRatio.roundToNearestPixel(5),
    paddingVertical: PixelRatio.roundToNearestPixel(5),
  },
  imageStyle: {
    height: 100,
    width: 100,
    borderRadius: 10,
  },
  imageInnerContainer: {
    position: 'relative',
    margin: 5,
    marginLeft: 20,
  },
  deleteIcon: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 10,
  },
});

export default React.memo(ImagesContainer);
