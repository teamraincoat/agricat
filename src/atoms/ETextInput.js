import React, {useState} from 'react';
import {View, Text, TextInput, StyleSheet} from 'react-native';
import {colors, styles} from '../styles';
import { hp, normalize } from '../styles/metrics';
import EButton from './EButton';

const ETextInput = ({
  email,
  phone,
  password,
  number,
  secure,
  error,
  errorText,
  style,
  label,
  ...props
}) => {


  const renderLabel = () => {
    return (
      <View>
        {label ? (
          <Text style={localStyles.labelStyle}>{label}</Text>
        ) : null}
      </View>
    );
  };


  const inputStyles = [
    localStyles.input,
    error && {borderColor: 'red'},
    style,
  ];
  const inputType = email
    ? 'email-address'
    : number
    ? 'numeric'
    : phone
    ? 'phone-pad'
    : 'default';

  return (
    <View style={[localStyles.container, styles.mt10]}>
       {renderLabel()}
      <TextInput
          style={inputStyles}
          autoComplete="off"
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType={inputType}
          underlineColorAndroid="transparent"
          placeholderTextColor={colors.grey}
          blurOnSubmit={false}
          {...props}
      />
      {errorText ? <Text style={localStyles.error}>{errorText}</Text> : null}
    </View>
  );
};

const localStyles = StyleSheet.create({
  container: {
    ...styles.mv5,
  },
  input: {
    color: colors.primary,
    height:hp(7),
    backgroundColor:colors.white,
    ...styles.h3,
    ...styles.mh20,
    ...styles.borderLight,
    ...styles.radius5,
    ...styles.mv5,
     ...styles.pv10,
     ...styles.ph15,
  },
  labelStyle:{
    color: colors.black,
    marginHorizontal: 20,
    ...styles.mh25,
    ...styles.h2,
    fontSize: normalize(12),
    },
  error: {
    color: colors.red,
    ...styles.mt5,
    ...styles.pl20,
    ...styles.pt10,
    ...styles.pl20,
  },
});

export default ETextInput;
