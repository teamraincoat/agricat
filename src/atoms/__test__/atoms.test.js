/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';

import EButton from '../EButton';
import BackgroundImage from '../BackgroundImage';
import CameraView from '../CameraView';
import EText from '../EText';
import ETextInput from '../ETextInput';
import ImagesContainer from '../ImagesContainer';

jest.useFakeTimers();
let mockedExp: jest.Mock;
jest.mock('react-native-iphone-x-helper', () => ({
  isIphoneX: () => 'You have called a mocked method 1!',
}));
jest.mock('../../assets/capture.png', () => 'CaptureImg');
jest.mock('react-native-camera-kit', () => ({
  CameraScreen: () => 'CameraScreen',
  CameraType: () => 'CameraType',
}));
jest.mock('react-native-blob-util', () => ({
  fs: {
    stat: () => 'fs.stat',
    readFile: () => 'fs.readFile',
  },
}));
jest.mock('react-native-image-resizer', () => ({
  createResizedImage: () => 'createResizedImage',
}));

jest.mock('../../localization/en.json', () => 'en');
jest.mock('../../localization/es.json', () => 'es');
jest.mock('react-hook-form', () => ({
  useForm: () => ({
    defaultValues: {
      maizeCultivation: [],
    },
    control: ({
      register, unregister, getValues, setValue, reset,
    }) => ({
      register,
      unregister,
      getValues,
      setValue,
      reset,
    }),
    formState: {
      errors: {},
    },
  }),
  Controller: () => 'Controller',
}));
jest.mock('@react-native-community/async-storage', () => ({
  AsyncStorage: {
    getItem: () => 'getItem',
    setItem: () => 'setItem',
    removeItem: () => 'removeItem',
  },
}));

jest.mock('react-native-localization', () => jest.fn().mockImplementation(() => ({
  init: mockedExp,
})));

test('render Button component', () => {
  const tree = renderer.create(<EButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render BackgroundImage component', () => {
  const tree = renderer.create(<BackgroundImage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render Camera component', () => {
  const { toJSON } = renderer.create(
    <CameraView
      setIsCameraVisible={() => {}}
      selectedFiles={[]}
      setSelectedFiles={() => {}}
    />,
  );
  expect(toJSON()).toMatchSnapshot();
});
test('render EText component', () => {
  const tree = renderer.create(<EText />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('render ETextInput component', () => {
  const tree = renderer.create(<ETextInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('render ImagesContainer component', () => {
  const tree = renderer
    .create(
      <ImagesContainer
        imageUri={{
          uri: undefined,
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
