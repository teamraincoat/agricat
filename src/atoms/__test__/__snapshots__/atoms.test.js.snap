// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render BackgroundImage component 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "height": 1334,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Image
    resizeMode="cover"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`render Button component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#242424",
      "borderRadius": 8,
      "color": "#ffffff",
      "elevation": 5,
      "height": 93,
      "justifyContent": "center",
      "marginTop": 15,
      "opacity": 1,
      "padding": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.84,
      "width": 675,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Proxima Nova",
          "fontSize": 36,
          "fontWeight": "700",
          "lineHeight": 20,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`render Camera component 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  CameraScreen
</View>
`;

exports[`render CameraView Component 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  CameraScreen
</View>
`;

exports[`render DropDownBtn component 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  CameraScreen
</View>
`;

exports[`render EText component 1`] = `
<Text
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`render ETextInput component 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <View />
  <TextInput
    autoCapitalize="none"
    autoComplete="off"
    autoCorrect={false}
    blurOnSubmit={false}
    editable={true}
    keyboardType="default"
    placeholderTextColor="#818181"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderWidth": 2,
          "color": "#000000",
          "fontFamily": "Proxima Nova",
          "fontSize": 28,
          "fontWeight": "400",
          "height": 93,
          "lineHeight": 17,
          "marginHorizontal": 20,
          "marginVertical": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`render ImagesContainer component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "flex-start",
      "marginVertical": 5,
      "paddingVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "margin": 5,
        "marginLeft": 20,
        "position": "relative",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 150,
          "width": 150,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "bottom": 10,
          "opacity": 1,
          "position": "absolute",
        }
      }
    >
      <RNSVGSvgView
        bbHeight={22}
        bbWidth={22}
        fill="none"
        focusable={false}
        height={22}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 22,
              "width": 22,
            },
          ]
        }
        width={22}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx={11}
            cy={11}
            fill={4294901760}
            propList={
              Array [
                "fill",
              ]
            }
            r={11}
          />
          <RNSVGPath
            d="M9.931 6.138a.38.38 0 0 0-.384.38H7.276a.38.38 0 1 0 0 .758h6.827a.379.379 0 1 0 0-.759h-2.27a.379.379 0 0 0-.385-.38H9.931ZM7.276 8.034v6.07c0 .419.34.758.758.758h5.31c.42 0 .76-.34.76-.759V8.034H7.275Z"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`render SectionCard component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderRadius": 5,
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 10,
        "width": 675,
      },
      undefined,
      Object {
        "backgroundColor": "#F9DFBB",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Proxima Nova",
        "fontSize": 24,
        "fontWeight": "400",
        "lineHeight": 17,
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Proxima Nova",
          "fontSize": 24,
          "fontWeight": "400",
          "lineHeight": 17,
          "marginVertical": 5,
          "textAlign": "center",
        },
        Object {
          "marginTop": 15,
        },
      ]
    }
  />
</View>
`;

exports[`render SectionHeader component 1`] = `
<View
  style={
    Object {
      "marginVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "justifyContent": "center",
        "width": 675,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Object {
            "alignSelf": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Proxima Nova",
              "fontSize": 48,
              "fontWeight": "700",
              "lineHeight": 29,
              "marginRight": 225,
            },
            Object {
              "marginRight": 0,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#000000",
          "fontFamily": "Proxima Nova",
          "fontSize": 28,
          "fontWeight": "400",
          "lineHeight": 17,
          "marginVertical": 10,
        }
      }
    />
  </View>
</View>
`;
